tests/expressions/tests.py

```diff
@@ -1043,6 +1043,38 @@ def test_duration_with_datetime(self):
         ).order_by('name')
         self.assertQuerysetEqual(over_estimate, ['e3', 'e4'], lambda e: e.name)
 
+    @skipUnlessDBFeature('supports_microsecond_precision')
+    def test_negative_timedelta_with_microsecond_support_update(self):
+        # subtract 30 microseconds, 30 seconds, 30 minutes, 2 hours and 2 days
+        Experiment.objects.filter(name='e0').annotate(
+            new_start=F('start') + datetime.timedelta(microseconds=-30),
+        ).annotate(
+            new_start=F('new_start') + datetime.timedelta(seconds=-30),
+        ).annotate(
+            new_start=F('new_start') + datetime.timedelta(minutes=-30),
+        ).annotate(
+            new_start=F('new_start') + datetime.timedelta(hours=-2),
+        ).annotate(
+            new_start=F('new_start') + datetime.timedelta(days=-2),
+        ).update(start=F('new_start'))
+        e0 = Experiment.objects.get(name='e0')
+        self.assertEqual(e0.start, datetime.datetime(2010, 6, 23, 9, 45, 0, 746970))
+
+    @skipIfDBFeature('supports_microsecond_precision')
```

Comment:
Possibly a single test could be used with some if statements to include or skip microsecond verification?